### ISSUE TITLE
Create ESLint rule

### ISSUE THREAD

**Issue by Odonno** [ngneat/until-destroy#134]
Is it possible to create a ESlint rule for the `untilDestroyed` operator? Which means that we check that every Observable in a Component should contain a `untilDestroyed(this)` and at the last position in the `pipe` chain.

**Comment by NetanelBasal** [comment]
You're welcome to create a PR. It'll be nice to validate that the decorator exists.

**Comment by NetanelBasal** [comment]
You can use [this](https://github.com/cartant/eslint-plugin-rxjs/blob/main/source/rules/no-unsafe-takeuntil.ts) as a reference.

**Comment by Odonno** [comment]
@NetanelBasal I believe that the `@UntilDestroy()` decorator is already checked at compilation. We could do it but what I really want to is detect Observable in every component and detect those who doesn't have the operator.

I already seen the work of @cartant but I have a hard time trying to understand how this thing works.

**Comment by NetanelBasal** [comment]
You are right, I forgot that we have protection.

**Comment by cartant** [comment]
FWIW, the rule linked above can be configured to use an alias - in addition to `takeUntil` - so you ought to be able to configure it to do what you want (if I understand you correctly). See [the docs for the rule](https://github.com/cartant/eslint-plugin-rxjs/blob/main/docs/rules/no-unsafe-takeuntil.md#options).

Also, something to be aware of is that ESLint rules are composable - see [`eslint-rule-composer`](https://github.com/not-an-aardvark/eslint-rule-composer) and [this post of mine](https://ncjamieson.com/til-overriding-a-frozen-object/). So if what you want to do can be represented by feeding configuration into an existing rule - i.e. to reduce boilerplate from more general, verbose config - that's possible.

**Comment by NetanelBasal** [comment]
Thanks, @cartant.

@Odonno I think that you want the following:

```ts
{
  "rxjs/no-unsafe-takeuntil": [
    "error",
    {
      "alias": ["untilDestroyed"]
    }
  ]
}
```

Is that right?

**Comment by Odonno** [comment]
Well, kinda. The only missing part would be to also check Observable without `pipe`. @cartant Can you confirm you also check Observable without `pipe`.

**Comment by NetanelBasal** [comment]
@Odonno I think that the rule only works if you are using the operator. It doesn't check if you have it.

**Comment by cartant** [comment]
[This rule](https://github.com/cartant/eslint-plugin-rxjs-angular/blob/main/docs/rules/prefer-takeuntil.md#options) is probably what you are looking for. The other just makes sure that the `takeUntil` - or alias - is in the right place. You probably want both, TBH. If `prefer-takeUntil` doesn't effect a failure if `subscribe` is called without `pipe` being used, that would be a bug.

**Comment by NetanelBasal** [comment]
Awesome rules. @Odonno you can use it. It'll be great if you can create a PR that adds this information to our README. 

Closing the issue. Thanks.

**Comment by mattdsteele** [comment]
@Odonno @NetanelBasal @cartant Are you successfully using this rule? I've attempted to install it in a project, but I'm unable to get the rule to fire as documented.

We've configured the plugin as such:

```json
"rxjs-angular/prefer-takeuntil": [
    "error", {
​​​     "alias": ["untilDestroyed"],
     "checkComplete": true,
     "checkDecorators": ["Component"],
     "checkDestroy": true
}​​​​ ]
```

And using code similar to this in our components:

```ts
this.xyzService
  .pipe(untilDestroyed(this))
  .subscribe(data => {	
  });
```

But the plugin throws errors for `Forbids calling 'subscribe' without an accompanying 'takeUntil'.

Any idea what might be going on? I'm happy to open a new ticket (here or in the rxjs-angular repo) if you'd prefer.
